### PR TITLE
For #2899 - Scroll to Collections on change, animate new collections

### PR DESCRIPTION
**Pull request by ekager**
### Pull Request checklist

- [meh] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -732,18 +739,56 @@ class HomeFragment : Fragment(), CoroutineScope, AccountObserver {
         emitAccountChanges()
     }
 
+    private fun scrollToCollections(addedTabsSize: Int) {
+        launch(Dispatchers.Main) {
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: Using Dispatchers.Main directly like this means there is no coroutineScope Job to handle cancellation. The fragment has a coroutineContext using Main already that will cancel and the delay() method will cooperatively cancel when it does so. Is there a reason not to use the current context and to create a new coroutineScope instead?

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: Does this change work @colintheshots ?

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 2]
```diff
@@ -481,6 +484,10 @@ class HomeFragment : Fragment(), CoroutineScope, AccountObserver {
                         }
                     }
                 }
+                launch(Dispatchers.Main) {
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: Please remove this dispatcher override here. Once this is done, I think we're ready to merge.

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: Oops missed this one. Got it!

**Pull request merged**